### ISSUE TITLE
MediaWiki?

### ISSUE THREAD

**Issue by GeekyStudios** [GeekyStudios/rFTC-wiki#18]
Writing this wiki had us find that Reddit-Markdown and Github are not the most friendly things for wiki creation. shurik179 [has suggested](https://github.com/GeekyStudios/rFTC-wiki/issues/14) converting this project into an actual wiki system, thereby moving all work to MediaWiki and Abandoning the Github.

shurik is offering their team's hosting. Since I also have a host with a few extra website spaces, I want to compare hosting plans.

Currently, I'm using Namecheap's shared Value Hosting, which includes:

 - 20GB Co-SSD Space (Shared with database)
 - 300,000 file limit
 - 50 databases
 - 500 database connections
 - 50 Email accounts
 - Unlimited email forwarders
 - 2 Shared CPU cores
 - 256 GB Shared RAM
 - Unlimited Bandwidth
 - cPanel (of course)

Naturally, I added HTTPS through Cloudflare.

**Comment by shurik179** [comment]
I do not expect the wiki will get too much traffic - maybe on the order of 100s visits a day, which any provider can manage (and we do not need 50 DBs). The only important consideration right now is reliability.

**Comment by GeekyStudios** [comment]
Yeah, I'm going to start up [a MediaWiki install](https://geekystudios.us/ftcwiki) on my host. My funding strategy for the domain is nonexistent as our team's only sponsor is a Chinese restaurant. Hopefully, I can get some funds by Christmas.

PS: 50 DBs can be very useful when the programmer of your Eng Notebook does not know how to use tables.

**Comment by MeRPG** [comment]
I think I have the same hosting plan as you @GeekyStudios. In addition, I'll take care of the domain sometime in the coming days. It might make sense to just host the mediawiki on my namecheap hosting, I'll look more into it this weekend.

**Comment by shurik179** [comment]
Our team would be happy to pay for domain name. As I said, we can create a media wiki install on our hosting as well. I would prefer to have some consensus before actually going ahead - we probably should invite Weznon to this discussion.
Should we agree on what domain name we should choose? For example, ftcwiki.org is available.

**Comment by GeekyStudios** [comment]
Yeah, [ftcwiki.org](https://www.namecheap.com/domains/registration/results.aspx?domain=ftcwiki.org) is good, I was planning to use it. I think we all want to use our individual hosting plans, and since I'm really bad at choosing things, why not luck of the draw? Adding @Weznon to participants.

**Comment by Weznon** [comment]
I don't see anything wrong with using mediawiki instead of reddit wikis, definitely is more powerful.

ftcwiki.org sounds good to me

**Comment by shurik179** [comment]
I've registered domain name ftcwiki.org, so far for one year
I created a new account on godaddy.com for this registration - so I can give control of this domain to other people, e.g. to /r/ftc moderators, for safekeeping. 

Let us decide on which hosting we use - we could use GeekyStudio hosting. Once the MediaWiki is installed and configured (I could help if needed - I've done it before, though I am not a real expert), I can then point the domain to that installation.

**Comment by GeekyStudios** [comment]
Namecheap's hosting is rather picky when it comes to using domains from 3rd parties. I'll have to manually edit some things and see if it works out. While we're at it, why not use Let's Encrypt or Cloudflare to provide SSL? I don't know how absurd this stack will look, but it's starting to look weird.

**Comment by Weznon** [comment]
im going to let you all figure out the hosting, cuz i have no idea how that works.

Once the media wiki stuff is setup i can help transferring pages and updating the wiki

**Comment by shurik179** [comment]
I have no experience with SSL, so I'll leave it to GeekyStudios.
I do have experience setting hosting and domain name from different providers/registrars - in fact, there are some reasons why I prefer that setup. Let me know if you need my help. Typically, there is little configuration needed on hosting side, and on domain name registrar, you just need to set  nameservers: https://www.godaddy.com/help/change-nameservers-for-my-domains-664

**Comment by GeekyStudios** [comment]
Sorry, school's getting a little heavy before winter break. Today's the last day, so I'll try to start coordinating the setup ASAP.

**Comment by GeekyStudios** [comment]
Sorry I'm a day late,
@shurik179, I think you need to direct your domain to [Namecheap nameservers](https://www.namecheap.com/support/knowledgebase/article.aspx/9518/32/how-to-set-up-our-shared-hosting-nameservers-for-domains-registered-with-godaddy) first. When the site is done configuring, I'll have to swap it to Cloudflare. I'll use my hosting package for this.

**Comment by shurik179** [comment]
I've updated the nameservers. They say the change can take up to 48 hrs to propagate

**Comment by GeekyStudios** [comment]
ftcwiki.org is running a MediaWiki instance. I'll add the Relic Recovery logo, add a few themes. I really don't know what else to do with MediaWiki. Anything else?

**Comment by shurik179** [comment]
One thing I'd do now is block users self-registration, so that only admins can register new users - otherwise, the moment we announce this wiki on reddit, you will get hundreds of bots per day registering. 

This can be done by adding
 
# Prevent new user registrations except by sysops
$wgGroupPermissions['*']['createaccount'] = false;
$wgGroupPermissions['*']['edit'] = false;

to LocalSettings.php in wiki root directory.

Next, I'd customize the sidebar - instructions on doing this are here: https://www.mediawiki.org/wiki/Manual:Interface/Sidebar#Customize_the_sidebar

I've created account for myself on the site - is it OK if I start editing?

**Comment by GeekyStudios** [comment]
Yeah, you can start editing. I'll do everything above.

**Comment by GeekyStudios** [comment]
I've replaced most of the WikiMarkup with Markdown. So a few pages are broken. hehe.

**Comment by Weznon** [comment]
is the admin registration of new users active? or am i just dumb cuz i cant find an option to make an account.

if it is active how would i get an account?

**Comment by shurik179** [comment]
GeekyStudios:
do you have full list of markup changes you made? It is difficult to create and edit pages without knowing the markup conventions. They need to be documented.

**Comment by GeekyStudios** [comment]
Hopefully, you all realize I'm a day late for most things. Weznon, I've sent an email for you to change your user password to waterwasp55555@gmail.com. shurik, I'm using the [Markdown MediaWiki Extension](https://github.com/bharley/mw-markdown) with [Markdown Extra syntax](https://michelf.ca/projects/php-markdown/extra/).  You can edit in markup if you add `{{WIKI}}` to the top of the document.

**Comment by shurik179** [comment]
I am working on moving pages to the wiki. Here are two issues I have encountered:
1. Can you enable file uploads, for images? 
https://www.mediawiki.org/wiki/Manual:Configuring_file_uploads
2. I tried using Wiki markup, by putting {{WIKI}} at top of the page. It works, but also produces words 
Template:WIKI
at the top of the page, as you can see here:
https://ftcwiki.org/index.php?title=Actobotics
Is there a way to get rid of this?

**Comment by GeekyStudios** [comment]
I've changed your permissions so that you *should* be in the sysop group now. File uploads should be enabled. Template:WIKI no longer shows as the page has been created.
